schemas/2018-04-01/Microsoft.Devices.json

```diff
@@ -0,0 +1,1005 @@
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "The number of partitions for receiving device-to-cloud messages in the Event Hub-compatible endpoint. See: https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messaging#device-to-cloud-messages."
+        }
+      },
+      "description": "The properties of the provisioned Event Hub-compatible endpoint used by the IoT hub."
+    },
+    "FallbackRouteProperties": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the route. The name can only include alphanumeric characters, periods, underscores, hyphens, has a maximum length of 64 characters, and must be unique."
+        },
+        "source": {
+          "oneOf": [
+            {
+              "type": "string",
+              "enum": [
+                "DeviceMessages"
+              ]
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "The source to which the routing rule is to be applied to. For example, DeviceMessages"
+        },
+        "condition": {
+          "type": "string",
+          "description": "The condition which is evaluated in order to apply the fallback route. If the condition is not provided it will evaluate to true by default. For grammar, See: https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-query-language"
+        },
+        "endpointNames": {
+          "oneOf": [
+            {
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "The list of endpoints to which the messages that satisfy the condition are routed to. Currently only 1 endpoint is allowed."
+        },
+        "isEnabled": {
+          "oneOf": [
+            {
+              "type": "boolean"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "Used to specify whether the fallback route is enabled."
+        }
+      },
+      "required": [
+        "source",
+        "endpointNames",
+        "isEnabled"
+      ],
+      "description": "The properties of the fallback route. IoT Hub uses these properties when it routes messages to the fallback endpoint."
+    },
+    "FeedbackProperties": {
+      "type": "object",
+      "properties": {
+        "lockDurationAsIso8601": {
+          "type": "string",
+          "format": "duration",
```

Comment:
>duration [](start = 21, length = 8)

duration format is not supported by Visual Studio. I guess you can remove them.

Reply:
Ack.